### DISCUSSION TITLE
Force refresh of all apps

### DISCUSSION THREAD

**Discussion by svend** [argoproj/argo-cd#6037] [Q&A]
What is the best way to ensure that all applications are refreshed?

After adding a label to `﻿﻿ignoreDifferences` most apps still showed `OutOfSync`. Manually syncing all of them is tedious, so I tried doing a rolling restart of the ArgoCD components, including the `server`, `application-controller`, and `redis`, but I couldn't clear the app cache.

(We have `application-controller`'s `--app-resync` set to 1 hour, so the apps should eventually refresh.)

**Comment by jannfis** [comment]
Hey @svend,

I would assume that - since `ignoreDifferences` will affect the reconciliation state of your apps independently of your source manifests - a refresh of the source manifests is not required for this setting to get effect. Usually, it just adapts the diffing once you changed the settings. So, maybe, the `ignoreDifferences` is not targeting the resources that you would expect it to.

However, to answer your question: You can use a _hard refresh_ of any application to force rebuilding the manifests from source. This can be either done using the CLI (`argocd app get --hard-refresh <appname>`), the UI (via the dropdown on the "Refresh" button in the application details view) or by sticking the annotation `argocd.argoproj.io/refresh: hard` on an `Application` CR in your cluster.

For targeting multiple applications with the CLI, you could use a small script like the following, given that you have also `jq` installed. Below example targets all apps associated to AppProject `default`:

```shell
for app in $(argocd app list -p default -o json | jq -r .[].metadata.name); do argocd app get --hard-refresh $app; done
```
